polaris-core/src/main/java/org/apache/polaris/core/config/FeatureConfiguration.java

```diff
@@ -301,8 +301,11 @@ public static void enforceFeatureEnabledOrThrow(
       PolarisConfiguration.<Boolean>builder()
           .key("OPTIMIZED_SIBLING_CHECK")
           .description(
-              "When set, an index is used to perform the sibling check between tables, views, and namespaces. Only"
-                  + " supported by the JDBC metastore.")
+              "When set, an index is used to perform the sibling check between tables, views, and namespaces. New " +
+                  "locations will be checked against previous ones based on components, so the new location " +
+                  "/foo/bar/ will check for a sibling at /, /foo/ and /foo/bar/%. In order for this check to " +
```

Comment:
nit: It might be preferable to avoid using plain file locations in example since the FILE storage type is disabled by default.